lib/entity/civimelder/Activiteit.php

```diff
@@ -0,0 +1,314 @@
+			$this->deelnemers[] = $deelnemer;
+			$deelnemer->setActiviteit($this);
+		}
+
+		return $this;
+	}
+
+	public function removeDeelnemer(Deelnemer $deelnemer): self {
+		if ($this->deelnemers->contains($deelnemer)) {
+			$this->deelnemers->removeElement($deelnemer);
+			// set the owning side to null (unless already changed)
+			if ($deelnemer->getActiviteit() === $this) {
+				$deelnemer->setActiviteit(null);
+			}
+		}
+
+		return $this;
+	}
+
+	// Eigenschappen
+	public function getTitel(): string {
+		return $this->getRawTitel() ?: $this->getReeks()->getRawTitel();
+	}
+
+	public function getBeschrijving(): string {
+		return $this->getRawBeschrijving() ?: $this->getReeks()->getRawBeschrijving();
+	}
+
+	public function getCapaciteit(): int {
+		return $this->getRawCapaciteit() ?: $this->getReeks()->getRawCapaciteit();
+	}
+
+	public function getRechtenAanmelden(): string {
+		return $this->getRawRechtenAanmelden() ?: $this->getReeks()->getRawRechtenAanmelden();
+	}
+
+	public function getRechtenLijstBekijken(): string {
+		return $this->getRawRechtenLijstBekijken() ?: $this->getReeks()->getRawRechtenLijstBekijken();
+	}
+
+	public function getRechtenLijstBeheren(): string {
+		return $this->getRawRechtenLijstBeheren() ?: $this->getReeks()->getRawRechtenLijstBeheren();
+	}
+
+	public function getMaxGasten(): int {
+		return $this->getRawMaxGasten() ?: $this->getReeks()->getRawMaxGasten();
+	}
+
+	public function getMaxAantal(): int {
+		return $this->getMaxGasten() + 1;
+	}
+
+	public function isAanmeldenMogelijk(): bool {
+		return $this->isRawAanmeldenMogelijk() ?: $this->getReeks()->isRawAanmeldenMogelijk();
+	}
+
+	public function getAanmeldenVanaf(): ?int {
+		return $this->getRawAanmeldenVanaf() ?: $this->getReeks()->getRawAanmeldenVanaf();
+	}
+
+	public function getAanmeldenTot(): ?int {
+		return $this->getRawAanmeldenTot() ?: $this->getReeks()->getRawAanmeldenTot();
+	}
+
+	public function isAfmeldenMogelijk(): bool {
+		return $this->isRawAfmeldenMogelijk() ?: $this->getReeks()->isRawAfmeldenMogelijk();
+	}
+
+	public function getAfmeldenTot(): ?int {
+		return $this->getRawAfmeldenTot() ?: $this->getReeks()->getRawAfmeldenTot();
+	}
+
+	// Tijden afmelden
+	private function getTijdVoor(int $minutes): DateTimeImmutable {
+		/** @noinspection PhpUnhandledExceptionInspection Minuten is altijd aantal minuten als integer */
+		$tijd = new DateInterval('PT' . $minutes . 'M');
+		return $this->getStart()->sub($tijd);
+	}
+
+	public function getStartAanmelden(): DateTimeImmutable {
+		return $this->getTijdVoor($this->getAanmeldenVanaf());
+	}
+
+	public function getEindAanmelden(): DateTimeImmutable {
+		return $this->getTijdVoor($this->getAanmeldenTot());
+	}
+
+	public function getEindAfmelden(): DateTimeImmutable {
+		return $this->getTijdVoor($this->getAfmeldenTot());
+	}
+
+	// Aanmeldingen
+	public function getAantalAanmeldingen(): int {
+		return $this->deelnemerRepository()->getAantalAanmeldingen($this);
```

Comment:
Kan je hier niet gewoon `$this->deelnemers->count()` doen? Deelnemers wordt al eager geladen toch?

Reply:
Geïnspireerd door maaltijdsysteem. Punt zit hem dat één tabel entry meerdere gasten kan zijn, dus hij doet een SUM i.p.v. count.